src/robot.coffee

```diff
@@ -376,17 +376,33 @@ class Robot
   #
   # Returns nothing.
   loadExternalScripts: (packages) ->
-    @logger.debug "Loading external-scripts from npm packages"
     try
       if packages instanceof Array
-        for pkg in packages
-          require(pkg)(@)
+        @loadScriptPackage(pkg) for pkg in packages
       else
-        for pkg, scripts of packages
-          require(pkg)(@, scripts)
+        @loadScriptPackage(pkg, scripts) for pkg, scripts of packages
+
+  # Public: load a script from an npm module
+  #
+  # pkg - A string of the package to load, or a script that has been required, but hasn't been loaded yet
+  # scripts - An Array of scripts to load from the package (optional)
+  #
+  # Examples:
+  #   robot.loadScriptPackage require('hubot-help')
+  #   robot.loadScriptPackage 'hubot-help'
+  #
+  # Note:
+  #   Specifying package as string may fail depending on your NODE_PATH settings.
+  #   This can be fixed by requiring the package, and passing that to this function.
+  #
+  # Returns nothing
+  loadScriptPackage: (pkg, scripts) ->
```

Comment:
Shouldn't we also be using this for `loadHubotScripts` and potentially for `loadFile`? Consolidating all calls to `require` would make the `new` refactor a lot easier as well as provide a consistent interface for programatic loading of any type of script.